### PR TITLE
fix(core): Add URL encoding to query params

### PR DESCRIPTION
**Pull request by kjurkovic**

### CODE REVIEW

File: traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java [hunk 1]
```diff
@@ -587,15 +589,15 @@ private URL getURL() {
 
                         sb.append(param.first);
                         sb.append("=");
-                        sb.append(param.second);
+                        sb.append(URLEncoder.encode(param.second, "UTF-8"));
```
Comment (smileman) [inline traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java]: In Router.java we use 
`Uri.encode()` a couple of times to encode dates.

Let's stick to one solution: either `URLEncoder.encode()` or `Uri.encode()`

And if you add encoring here - it's better to remove all date encodings above in this file.

Comment (kjurkovic) [reply traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java]: I haven't notice dates encoding. I think it's better to do it on one place, so I've removed it from dates and added here with `Uri.encode()`

File: traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java [hunk 2]
```diff
@@ -51,8 +51,10 @@
 import org.json.JSONException;
 import org.json.JSONObject;
 
+import java.io.UnsupportedEncodingException;
```
Comment (smileman) [inline traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java]: unused import

File: traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java [hunk 3]
```diff
@@ -51,8 +51,10 @@
...
 import java.net.MalformedURLException;
 import java.net.URL;
+import java.net.URLEncoder;
```
Comment (smileman) [inline traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java]: unused import

Comment (kjurkovic) [reply traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java]: I'm actually pretty amazed that you notice stuff like this. Fixed ;)

File: build.gradle [hunk 4]
```diff
@@ -10,7 +10,7 @@ buildscript {
     ext.app_center_sdk_version = '1.11.0'
     ext.stripe_sdk_version = '10.4.6'
     ext.material_library_version = '1.0.0'
-    ext.traveler_core_version = '0.20.1'
+    ext.traveler_core_version = '0.21.0'
```
Comment (OmarPadierna) [inline build.gradle]: why are you downgrading the version here?

Comment (kjurkovic) [reply build.gradle]: @OmarPadierna upgrading -> `0.21.0 > 0.20.1`

Comment (amirziarati) [reply build.gradle]: whenever you changed core you push your changes and then when the changes are merged to master a new core build is ready to use. then you ping me and i push a new ui sdk with updated core version or you can make a branch change the ui core version to latest and send a pr.

Comment (kjurkovic) [reply build.gradle]: Ok, I removed this update. Only fix for encoding is in PR now

File: traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java (hunk 3 above)
Comment (amirziarati) [reply traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/Router.java]: @kjurkovic use F2 in android studio in your changed files to jump to lint errors and try to send them without lint error :D although i myself forget to do it some times and Gennady finds my lint errors :D

**Pull request merged**